### PR TITLE
SONARJS-729 Do not include complexity of nested functions into nestin…

### PR DESCRIPTION
**Pull request by ivandalbosco**
…g one complexity

### CODE REVIEW

File: javascript-frontend/src/test/java/org/sonar/javascript/metrics/MetricsTest.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
   public void complexity() {
     String path = "src/test/resources/metrics/complexity.js";
     Tree tree = p.parse(new File(path));
-    assertThat(new ComplexityVisitor().getComplexity(tree)).isEqualTo(29);
+    assertThat(new ComplexityVisitor().getComplexity(tree)).isEqualTo(22);
```
Comment (vilchik-elena) [inline javascript-frontend/src/test/java/org/sonar/javascript/metrics/MetricsTest.java]: @ivandalbosco this number should be the same: it's file complexity

File: javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java [hunk 2]
```diff
@@ -47,47 +47,67 @@
 public class ComplexityVisitor extends DoubleDispatchVisitor {
 
   private List<Tree> complexityTrees;
+
   private Set<Tree> excludedReturns;
 
+  private boolean isNested;
```
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java]: @ivandalbosco i would rename it to `insideFunction`

File: javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java [hunk 3]
```diff
@@ -47,47 +47,67 @@
...
     scan(tree);
     return this.complexityTrees;
   }
 
   @Override
   public void visitMethodDeclaration(MethodDeclarationTree tree) {
-    add(tree.name());
-    excludeLastReturn(tree.body().statements());
-    super.visitMethodDeclaration(tree);
+    if (!isNested) {
```
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java]: @ivandalbosco is it possible to refactor this 6 lines?

File: javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java (hunk 2 above)
Comment (ivandalbosco) [inline javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java]: @vilchik-elena OK

File: javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java (hunk 3 above)
Comment (ivandalbosco) [inline javascript-frontend/src/main/java/org/sonar/javascript/metrics/ComplexityVisitor.java]: @vilchik-elena I tried, didn't find.

File: javascript-frontend/src/test/java/org/sonar/javascript/metrics/MetricsTest.java (hunk 1 above)
Comment (ivandalbosco) [inline javascript-frontend/src/test/java/org/sonar/javascript/metrics/MetricsTest.java]: @vilchik-elena Thanks! Done

**Comment by ivandalbosco** [comment]
Closing again, because the QA tests were not started.

**Pull request closed without merging**